video/out/opengl/context_drm_egl.c

```diff
@@ -388,27 +388,17 @@ static void wait_fence(struct ra_ctx *ctx)
     }
 }
 
-static bool drm_egl_start_frame(struct ra_swapchain *sw, struct ra_fbo *out_fbo)
-{
-    struct ra_ctx *ctx = sw->ctx;
-    struct priv *p = ctx->priv;
-    struct vo_drm_state *drm = ctx->vo->drm;
-
-    if (!drm->atomic_context->request) {
-        drm->atomic_context->request = drmModeAtomicAlloc();
-        p->drm_params.atomic_request_ptr = &drm->atomic_context->request;
-    }
-
-    return ra_gl_ctx_start_frame(sw, out_fbo);
-}
-
 static bool drm_egl_submit_frame(struct ra_swapchain *sw, const struct vo_frame *frame)
 {
     struct ra_ctx *ctx = sw->ctx;
     struct vo_drm_state *drm = ctx->vo->drm;
 
     drm->still = frame->still;
 
+    // With gpu-next, only the drm state needs to be updated.
+    if (vo_is_gpu_next(ctx->vo))
```

Comment:
I don't like that we split the logic of this module based of underlying VO driver used. I don't know, couldn't code decide on this in more structured way? For why shouldn't this be done in `ra_gl_ctx_submit_frame` itself?

Reply:
I mean I definitely dislike it too. We could certainly move it to `ra_gl_ctx_submit_frame` if you want but that feels just as unstructured imo. The reality is simply that vo_gpu_next doesn't need `ra_gl_ctx_submit_frame` (in fact it doesn't need `ra_gl_ctx_start_frame` either really) but due to the goofy external swapchain implementation of drm, it needs to be called for the sole purpose of flipping one bool which feels dumb.

I do plan to try out a total rewrite of this, but I didn't want to mix it in with the bug fix.

Reply:
I trust you on this. I just wanted to point out without clear module separation we invite more workarounds.